Enable new feedback system.

BillWagner opened the pull request:
Fixes #37025

Enable the new feedback system.

[Internal preview link](https://review.learn.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-12?branch=pr-en-us-37044)

You can try this on any page, but you'll need the PR branch, and the extra query string parameter `preview=open-source-feedback` to see the new experience.

BillWagner committed:
Enable new feedback system.

Fixes #37025

Enable the new feedback system.

homing1 commented on docfx.json:
```diff
@@ -126,6 +126,14 @@
             "featureFlags": [
                 "enable_uhf_ppe"
             ],
+            "open_source_feedback": {
```
```suggestion
"open_source_feedback_contributorGuideUrl": "https://learn.microsoft.com/contribute/content/dotnet/dotnet-contribute",
"open_source_feedback_issueTemplateName": "customer-feedback",
"open_source_feedback_productName": ".NET",
"open_source_feedback_productDescription": "Free. Cross platform. Open source. A developer platform for building all your apps: web, mobile, desktop, gaming, IoT, and more. Supported on Windows, Linux, and macOS.",
"open_source_feedback_productLogoLightUrl": "https://learn.microsoft.com/media/logos/logo_net.svg",
"open_source_feedback_productLogoDarkUrl": "https://learn.microsoft.com/media/logos/logo_net.svg",
```

BillWagner committed:
use updated config data.

homing1 commented on docfx.json:
```diff
@@ -126,6 +126,12 @@
             "featureFlags": [
                 "enable_uhf_ppe"
             ],
+            "contributorGuideUrl": "https://learn.microsoft.com/contribute/content/dotnet/dotnet-contribute",
```
Please add the `open_source_feedback_` prefix to the fields.

open_source_feedback_contributorGuideUrl
open_source_feedback_issueTemplateName
open_source_feedback_productName
open_source_feedback_productDescription
open_source_feedback_productLogoLightUrl
open_source_feedback_productLogoDarkUrl

BillWagner replied:
Thanks @homing1 I mis-read your previous comment.

In the next commit, I'll also move this to the file meta data section.

homing1 commented on docfx.json:
```diff
@@ -126,6 +126,12 @@
+            "productDescription": "Free. Cross platform. Open source. A developer platform for building all your apps: web, mobile, desktop, gaming, IoT, and more. Supported on Windows, Linux, and macOS.",
+            "productLogoLightUrl": "https://learn.microsoft.com/media/logos/logo_net.svg",
+            "productLogoDarkUrl": "https://learn.microsoft.com/media/logos/logo_net.svg",
             "feedback_system": "GitHub",
```
```suggestion
"feedback_system": "OpenSource",
```

BillWagner committed:
respond to feedback

- Move to the `fileMetadata` section.
- Update element names.

BillWagner committed:
short product blurb

The product blurb was being shortened with ellipses, so use a shorter one.

BillWagner committed:
add yml files for custom feedback.

BillWagner commented:
closing and reopening for a fresh build

BillWagner committed:
tweak the file paths.

BillWagner committed:
add yml extension instead of *.*

I had tried getting a glob pattern of all files for some of the metadata. That doesn't seem to work.

BillWagner committed:
missed one

IEvangelist commented on docfx.json:
```diff
@@ -162,12 +162,38 @@
         },
         "fileMetadata": {
             "feedback_system": {
+                "docs/**.md":  "OpenSource",
+                "docs/**.yml": "OpenSource",
```
```suggestion
                "docs/**.*":  "OpenSource",
```

IEvangelist commented on docfx.json:
```diff
@@ -162,12 +162,38 @@
                 "docs/framework/wcf/**/**.md": "None"
             },
+            "open_source_feedback_contributorGuideUrl": {
+                "docs/**/**.md":  "https://learn.microsoft.com/contribute/content/dotnet/dotnet-contribute",
+                "docs/**/**.yml": "https://learn.microsoft.com/contribute/content/dotnet/dotnet-contribute"
```
```suggestion
                "docs/**/**.*":  "https://learn.microsoft.com/contribute/content/dotnet/dotnet-contribute"
```

IEvangelist commented on docfx.json:
```diff
@@ -162,12 +162,38 @@
+                "docs/**/**.yml": "https://learn.microsoft.com/contribute/content/dotnet/dotnet-contribute"
+            },
+            "open_source_feedback_issueTemplateName": {
+                "docs/**.md":  "customer-feedback",
+                "docs/**.yml": "customer-feedback"
```
```suggestion
                "docs/**.*":  "customer-feedback"
```

IEvangelist commented on docfx.json:
```diff
@@ -162,12 +162,38 @@
+                "docs/**.yml": "customer-feedback"
+            },
+            "open_source_feedback_productName": {
+                "docs/**.md":  ".NET",
+                "docs/**.yml": ".NET"
```
```suggestion
                "docs/**.*":  ".NET"
```

IEvangelist commented on docfx.json:
```diff
@@ -162,12 +162,38 @@
+                "docs/**.yml": ".NET"
+            },
+            "open_source_feedback_productDescription": {
+                "docs/**.md":  "Free. Cross platform. Open source. A developer platform for building all your apps.",
+                "docs/**.yml": "Free. Cross platform. Open source. A developer platform for building all your apps."
```
```suggestion
                "docs/**.*":  "Free. Cross platform. Open source. A developer platform for building all your apps."
```

IEvangelist commented on docfx.json:
```diff
@@ -162,12 +162,38 @@
+                "docs/**.yml": "Free. Cross platform. Open source. A developer platform for building all your apps."
+            },
+            "open_source_feedback_productLogoLightUrl": {
+                "docs/**.md":  "https://learn.microsoft.com/media/logos/logo_net.svg",
+                "docs/**.yml": "https://learn.microsoft.com/media/logos/logo_net.svg"
```
```suggestion
                "docs/**.*":  "https://learn.microsoft.com/media/logos/logo_net.svg"
```

IEvangelist commented on docfx.json:
```diff
@@ -162,12 +162,38 @@
+                "docs/**.yml": "https://learn.microsoft.com/media/logos/logo_net.svg"
+            },
+            "open_source_feedback_productLogoDarkUrl": {
+                "docs/**.md":  "https://learn.microsoft.com/media/logos/logo_net.svg",
+                "docs/**.yml": "https://learn.microsoft.com/media/logos/logo_net.svg"
```
```suggestion
                "docs/**.*":  "https://learn.microsoft.com/media/logos/logo_net.svg"
```

IEvangelist reviewed (commented):
Simplify this configuration such that it's for any published file.

BillWagner committed:
update for latest fixes

BillWagner committed:
template key needs the extension

BillWagner committed:
simplify glob pattern

IEvangelist commented on docfx.json:
```diff
@@ -162,12 +162,34 @@
+                "docs/**.*": "https://github.com/dotnet/docs/issues/new?template=customer-feedback.yml"
+            },
+            "open_source_feedback_productName": {
+                "docs/**.*": ".NET"
```
```suggestion
                "docs/**.*": ".NET feedback"
```

IEvangelist commented on docfx.json:
```diff
@@ -162,12 +162,34 @@
+                "docs/**.*": ".NET"
+            },
+            "open_source_feedback_productDescription": {
+                "docs/**.*": "Free. Cross platform. Open source. A developer platform for building all your apps."
```
```suggestion
                "docs/**.*": "Like the .NET product, our documentation is Open source. Provide feedback here."
```

BillWagner committed:
Apply suggestions from code review

Co-authored-by: David Pine <david.pine@microsoft.com>

BillWagner commented on docfx.json:
```diff
@@ -162,12 +162,34 @@
+                "docs/**.*": "The .NET documentation is open source. Provide feedback here."
+            },
+            "open_source_feedback_issueTitle": {
+                "docs/**.*": "[Customer issue] - "
```
Let's leave the title blank:

```suggestion
                "docs/**.*": ""
```

BillWagner committed:
Update docfx.json

BillWagner commented:
closing and reopening for a fresh build.
